### PR TITLE
Replace CV with PV for project queries

### PR DESCRIPTION
**Pull request by peripheralist**
## What does this PR do and why?

Adds support for new subgraph Project property `pv`. This property indicates the version of the _project_ (1, 1.1, 2), irrespective of the contracts version (`cv`: 1, 1.1, 2, 3) used for loading a specifics contract version in the UI.

Only UI change is the addition of a `V2` or `V3` label to project activity events using the JBETHPaymentTerminal for the pv2 project dashboard, indicating which version of the JBETHPaymentTerminal contract was used.

## Screenshots or screen recordings

image

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/utils/server/pages/props.ts [hunk 1]
```diff
@@ -59,18 +55,13 @@ export async function getProjectProps(
...
-
-    const cv = isV3ProjectResult ? CV_V3 : CV_V2
+    const [metadata] = await Promise.all([findProjectMetadata({ metadataCid })])
 
     return {
       props: {
         metadata,
         projectId,
-        cv,
+        contractsVersion, // How to get contractsVersion here? Is it needed?
```
Comment (peripheralist) [inline src/utils/server/pages/props.ts:64]: WIP, unsure if we need to have this property

File: src/pages/create/ProjectPreview.tsx [hunk 2]
```diff
@@ -90,7 +87,12 @@ export function ProjectPreview() {
 
   return (
     <ProjectMetadataContext.Provider
-      value={{ projectMetadata, isArchived: false, projectId: 0, cv }}
+      value={{
+        projectMetadata,
+        isArchived: false,
+        projectId: 0,
+        pv, // TODO unsure how to calcualte PV here. This page uses V2V3 hooks, but isn't V2V3 itself? If limited to V2V3, pv will always be '2'
```
Comment (peripheralist) [inline src/pages/create/ProjectPreview.tsx:94]: TODO

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-import { Button, Space, Statistic } from 'antd'
-import { ProjectMetadataContext } from 'contexts/projectMetadataContext'
-import { useContext, useState } from 'react'
+import { Button, Space } from 'antd'
+import { useState } from 'react'
+
 import { useIsUpgradeAvailable } from './hooks/useIsUpgradeAvailable'
 import { UpgradeWizard } from './UpgradeWizard/UpgradeWizard'
 
+// TODO should this page be removed?
```
Comment (peripheralist) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx:8]: unclear on this page. can we remove it?

File: src/components/ProjectSafeDashboard/TransactionCollapse.tsx [hunk 4]
```diff
@@ -40,10 +40,11 @@ export function TransactionCollapse({
   const [expanded, setExpanded] = useState<boolean>(Boolean(selected))
 
   let versionText: string | undefined
-  switch (version) {
-    case CV_V1:
-      versionText = 'V1'
-      break
+  switch (cv) {
+    // TODO why do we need '1' here on V2V3 component?
```
Comment (tomquirk) [inline src/components/ProjectSafeDashboard/TransactionCollapse.tsx:44]: I think this component is version agnostic, used in safe page for both v1 and v2

File: src/components/ProjectSafeDashboard/juiceboxTransactions/reconfigureFundingCyclesOf/index.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ export function ReconfigureFundingCyclesOfTransaction({
       transaction={transaction}
       title={t`Reconfigure funding cycle`}
       isPastTransaction={isPastTransaction}
-      version={txVersion}
+      cv={txVersion}
```
Comment (tomquirk) [inline src/components/ProjectSafeDashboard/juiceboxTransactions/reconfigureFundingCyclesOf/index.tsx:19]: **praise**

File: src/utils/v2v3/contractLoaders/JuiceboxV2OrV3.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 import { NetworkName } from 'models/network-name'
 import { V2V3ContractName } from 'models/v2v3/contracts'
+import { CV2V3 } from 'models/v2v3/cv'
 import { ContractJson } from '../loadV2V3Contract'
 import { loadJuiceboxV2Contract } from './JuiceboxV2'
 import { loadJuiceboxV3Contract } from './JuiceboxV3'
 
 export async function loadJuiceboxV2OrV3Contract(
-  cv: CV,
+  cv: CV2V3,
```
Comment (tomquirk) [inline src/utils/v2v3/contractLoaders/JuiceboxV2OrV3.ts:10]: **praise**

**Review by tomquirk — APPROVED** [review]

File: src/components/ProjectSafeDashboard/TransactionCollapse.tsx (hunk 4 above)
Comment (peripheralist) [reply src/components/ProjectSafeDashboard/TransactionCollapse.tsx]: oh wait, i could have sworn this was a V2V3 component... nvm

File: src/components/activityEventElems/PayEventElem.tsx [hunk 7]
```diff
@@ -62,12 +67,16 @@ export default function PayEventElem({
         </div>
 
         <div style={{ textAlign: 'right' }}>
-          {event.timestamp && (
-            <div style={smallHeaderStyle(colors)}>
-              {formatHistoricalDate(event.timestamp * 1000)}{' '}
-              <EtherscanLink value={event.txHash} type="tx" />
-            </div>
-          )}
+          <div style={smallHeaderStyle(colors)}>
```
Comment (tomquirk) [inline src/components/activityEventElems/PayEventElem.tsx:70]: **praise**

File: src/hooks/v1/transactor/TransferTokensTx.ts [hunk 8]
```diff
@@ -60,7 +60,7 @@ export function useTransferTokensTx(): TransactorInstance<{
         txOpts,
         missingParam,
         functionName: 'transfer',
-        cv,
+        cv: '1',
```
Comment (tomquirk) [inline src/hooks/v1/transactor/TransferTokensTx.ts:63]:
```suggestion
        cv: CV_V1,
```

File: src/hooks/V2V3TerminalVersion.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          `@jbx-protocol/contracts-v2-latest/deployments/${readNetwork.name}/JBETHPaymentTerminal.json`
+        )) as ContractJson
+      ).address
+
+      try {
+        setV2Address(address)
+      } catch (e) {
+        console.error('Failed to load V2 JBETHPaymentTerminal', e)
+      }
+    }
```
Comment (tomquirk) [inline src/hooks/V2V3TerminalVersion.ts:25]: **suggestion** to save some code, can probably get away with using the existing `loadJuiceboxV2Contract`, `loadJuiceboxV3Contract`, or `useLoadV2V3Contract` and specify the CV ([example](https://github.com/jbx-protocol/juice-interface/blob/76b135157034cca6936c413bebefa4a434e19c60/src/utils/safe.ts#L48-L55))

File: src/hooks/Transactor.ts [hunk 10]
```diff
@@ -196,7 +196,7 @@ export function handleTransactionException({
   txOpts?: TxOpts
   missingParam?: string
   functionName: string
-  cv: CV | undefined
+  cv: '1' | CV2V3 | undefined // TODO should this be CV?
```
Comment (tomquirk) [inline src/hooks/Transactor.ts:199]: **question** should we just revert this line?

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/hooks/reconfigureFundingCycle.ts [hunk 11]
```diff
@@ -118,7 +117,7 @@ export const useReconfigureFundingCycle = ({
       async onConfirmed() {
         if (projectId) {
           await revalidateProject({
-            cv: cv as CV2V3,
+            pv: '2',
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/hooks/reconfigureFundingCycle.ts:120]:
```suggestion
            pv: PV_V2,
```

File: src/hooks/v2v3/transactor/MintTokensTx.ts [hunk 12]
```diff
@@ -55,7 +56,7 @@ export function useMintTokensTx(): TransactorInstance<{
     } catch {
       const missingParam = !transactor
         ? 'transactor'
-        : !JBController
+        : JBController
```
Comment (tomquirk) [inline src/hooks/v2v3/transactor/MintTokensTx.ts:59]: **question** whats the reason for this change?

File: src/components/v2v3/V2V3Project/V2V3ProjectHeaderActions/ContractVersionSelect.tsx [hunk 13]
```diff
@@ -3,10 +3,10 @@ import { BaseOptionType } from 'antd/lib/select'
 import { ProjectVersionBadge } from 'components/ProjectVersionBadge'
 import { CV_V2, CV_V3 } from 'constants/cv'
 import { V2V3ContractsContext } from 'contexts/v2v3/V2V3ContractsContext'
-import { CV2V3 } from 'models/cv'
+import { CV2V3 } from 'models/v2v3/cv'
 import { useContext } from 'react'
 
-const CV_LABELS = {
+const CV_LABELS: Record<CV2V3, string> = {
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectHeaderActions/ContractVersionSelect.tsx:9]: **praise**

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx [hunk 14]
```diff
@@ -1,12 +1,12 @@
...
-import { useContext, useState } from 'react'
+import { Button, Space } from 'antd'
+import { useState } from 'react'
+
 import { useIsUpgradeAvailable } from './hooks/useIsUpgradeAvailable'
 import { UpgradeWizard } from './UpgradeWizard/UpgradeWizard'
 
 export function ProjectUpgradesPage() {
-  const { cv } = useContext(ProjectMetadataContext)
+  // const { cv } = useContext(ProjectMetadataContext)
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx:9]: **question** can probably keep this as-is, wdyt?

**Review by tomquirk — COMMENTED** [review]
@peripheralist high quality stuff ser <3, i left a few comments!

File: src/hooks/Transactor.ts (hunk 10 above)
Comment (peripheralist) [reply src/hooks/Transactor.ts:199]: oops this was corrected but should have removed the TODO

File: src/hooks/v2v3/transactor/MintTokensTx.ts (hunk 12 above)
Comment (peripheralist) [reply src/hooks/v2v3/transactor/MintTokensTx.ts:59]: thank youuu! dumb mistake

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx (hunk 14 above)
Comment (peripheralist) [reply src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ProjectUpgradesPage/ProjectUpgradesPage.tsx:9]: oh thanks for addressing, probably shouldn't have left this page like this

`cv` got removed from ProjectMetadataContext, and we don't have a stat for "Current version" that really makes sense here (pv only goes up to 2). adding a comment for context but i'll leave this as is as a reference while this changed cv property is new

**Pull request merged**